### PR TITLE
Storage 4.10 Rel Notes

### PR DESCRIPTION
**Pull request by lpettyjo**
4.10

**Preview**: https://deploy-preview-38258--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html

**PTAL**: @openshift/storage, @jhou1

### CODE REVIEW

**Comment by lpettyjo** [comment]
The build is failing because of xrefs to new CSI Driver Operators.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 1]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
 [id="ocp-4-10-storage"]
 === Storage
 
+[id="ocp-4-10-storage-alicloud-disk-csi-operator"]
+==== Persistent storage using the Alibaba AliCloud Disk CSI Driver Operator
+{product-title} is capable of provisioning PVs using the CSI driver for AliCloud Disk. The AliCloud Disk Driver Operator that manages this driver is generally available.
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:260]:
```suggestion
{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for AliCloud Disk. The AliCloud Disk Driver Operator that manages this driver is generally available.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 2]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc[AliCloud Disk CSI Driver Operator].
+
+[id="ocp-4-10-storage-azure-file-csi-operator"]
+==== Persistent storage using the Microsoft Azure File CSI Driver Operator (Technology Preview)
+{product-title} is capable of provisioning PVs using the CSI driver for Azure File. The Azure File Driver Operator that manages this driver is in Technology Preview.
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:266]:
```suggestion
{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Azure File. The Azure File Driver Operator that manages this driver is in Technology Preview.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 3]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure-file.adoc[Azure File CSI Driver Operator].
+
+[id="ocp-4-10-storage-ibm-cloud-csi-operator"]
+==== Persistent storage using the IBM Cloud CSI Driver Operator
+{product-title} is capable of provisioning PVs using the CSI driver for IBM Cloud. The IBM Cloud Driver Operator that manages this driver is generally available.
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:272]:
```suggestion
{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for IBM Cloud. The IBM Cloud Driver Operator that manages this driver is generally available.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 4]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+==== Automatic CSI migration supports Microsoft Azure file (Technology Preview)
+
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent CSI drivers became available as a Technology Preview feature. This feature now supports automatic migration for the Azure File in-tree plug-in to the Azure File CSI driver.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
+
+[id="ocp-4-10-storage-csi-migration-vsphere"]
+==== Automatic CSI migration supports VMware vSphere (Technology Preview)
+
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent CSI drivers became available as a Technology Preview feature. This feature now supports automatic migration for the vSphere in-tree plug-in to the vSphere CSI driver.
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:307]:
```suggestion
Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. This feature now supports automatic migration for the vSphere in-tree plug-in to the vSphere CSI driver.
```

**Review by michaelryanpeter — COMMENTED** [review]
I noted the places where PV and CSI weren't spelled out in each release note. 

Otherwise, this LGTM. I will switch the label to _Peer Review Done_, but please wait to merge until the xrefs clear and the Travis build passes.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 5]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
 [id="ocp-4-10-storage"]
 === Storage
 
+[id="ocp-4-10-storage-alicloud-disk-csi-operator"]
+==== Persistent storage using the Alibaba AliCloud Disk CSI Driver Operator
+{product-title} is capable of provisioning PVs using the CSI driver for AliCloud Disk. The AliCloud Disk Driver Operator that manages this driver is generally available.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc[AliCloud Disk CSI Driver Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:262]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc#<anchor_id>[AliCloud Disk CSI Driver Operator].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 6]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-azure-file-csi-operator"]
+==== Persistent storage using the Microsoft Azure File CSI Driver Operator (Technology Preview)
+{product-title} is capable of provisioning PVs using the CSI driver for Azure File. The Azure File Driver Operator that manages this driver is in Technology Preview.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure-file.adoc[Azure File CSI Driver Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:268]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure-file.adoc#<anchor_id>[Azure File CSI Driver Operator].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 7]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-ibm-cloud-csi-operator"]
+==== Persistent storage using the IBM Cloud CSI Driver Operator
+{product-title} is capable of provisioning PVs using the CSI driver for IBM Cloud. The IBM Cloud Driver Operator that manages this driver is generally available.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-ibm-cloud.adoc[IBM Cloud CSI Driver Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:274]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-ibm-cloud.adoc#<anchor_id>[IBM Cloud CSI Driver Operator].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 8]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-vsphere-csi-operator"]
+==== Persistent storage using VMware vSphere CSI Driver Operator is generally available
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for vSphere. This feature was previously introduced as a Technology Preview feature in {product-title} 4.8 and is now generally available and enabled by default in {product-title} 4.10.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc[vSphere CSI Driver Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:281]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#<anchor_id>[vSphere CSI Driver Operator].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 9]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-azure-disk-csi-operator"]
+==== Persistent storage using Microsoft Azure Disk CSI Driver Operator is generally available
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Azure Disk. This feature was previously introduced as a Technology Preview feature in {product-title} 4.8 and is now generally available and enabled by default in {product-title} 4.10.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure.adoc[Azure Disk CSI Driver Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:288]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure.adoc#<anchor_id>[Azure Disk CSI Driver Operator].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 10]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-aws-efs-csi-operator"]
+==== Persistent storage using AWS Elastic File Storage CSI Driver Operator is generally available
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for AWS Elastic File Storage (EFS). This feature was previously introduced as a Technology Preview feature in {product-title} 4.9 and is now generally available and enabled by default in {product-title} 4.10.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc[AWS EFS Disk CSI Driver Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:295]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc#<anchor_id>[AWS EFS Disk CSI Driver Operator].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 11]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-csi-migration-azure-file"]
+==== Automatic CSI migration supports Microsoft Azure file (Technology Preview)
+
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent CSI drivers became available as a Technology Preview feature. This feature now supports automatic migration for the Azure File in-tree plug-in to the Azure File CSI driver.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:302]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc#<anchor_id>[CSI automatic migration].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 12]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-csi-migration-vsphere"]
+==== Automatic CSI migration supports VMware vSphere (Technology Preview)
+
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent CSI drivers became available as a Technology Preview feature. This feature now supports automatic migration for the vSphere in-tree plug-in to the vSphere CSI driver.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:309]:
```suggestion
For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc#<anchor_id>[CSI automatic migration].
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 13]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-fsGroup-pod-timeouts"]
+==== Using fsGroup to reduce pod timeouts
+
+If a storage volume contains many files (~1,000,000 or greater), you may experience pod timeouts.
+
+{product-title} 4.10 introduces the ability to use `fsGroup` and `fsGroupChangePolicy` to skip recursive permission change for the storage volume, thus helping to avoid pod timeout problems.
+
+For more information, see xref:../storage/persistent_storage/understanding-persistent-storage#storage-persistent-storage-fsGroup.adoc[Using fsGroup to reduce pod timeouts].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:318]:
```suggestion
For more information, see xref:../storage/persistent_storage/understanding-persistent-storage.adoc#storage-persistent-storage-fsGroup[Using fsGroup to reduce pod timeouts].
```
With this one, you have an anchor, but I think your file extension is in the wrong place.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 14]
```diff
@@ -255,6 +255,75 @@ For more information, see xref:../networking/ingress-operator.adoc#nw-configurin
...
+
+[id="ocp-4-10-storage-shared-dirs-local-vols"]
+==== Using shared directories with local volumes
+
+You can now specify a directory to be consumed using the local volume as an alternative to using a path containing a list of local storage devices to choose from. A directory can only be used in `FileSystem volumeMode`.
+
+For more information, see xref:../storage/persistent_storage/configuring-persistent-storage#local-create-cr-manual_persistent-storage-local.adoc[Provisioning local volumes without the Local Storage Operator].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-10-release-notes.adoc:325]:
```suggestion
For more information, see xref:../storage/persistent_storage/configuring-persistent-storage.adoc#local-create-cr-manual_persistent-storage-local[Provisioning local volumes without the Local Storage Operator].
```
See the comment for line 318.

**Review by michaelryanpeter — COMMENTED** [review]
In my first pass, I missed that all `xrefs` need to have an anchor id at the end. Please reach out if you have questions.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 15]
```diff
@@ -286,6 +286,75 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
 [id="ocp-4-10-storage"]
 === Storage
 
+[id="ocp-4-10-storage-alicloud-disk-csi-operator"]
+==== Persistent storage using the Alibaba AliCloud Disk CSI Driver Operator
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for AliCloud Disk. The AliCloud Disk Driver Operator that manages this driver is generally available.
```
Comment (mburke5678) [inline release_notes/ocp-4-10-release-notes.adoc:291]: @lpettyjo I believe Alibaba Cloud is going as TP in 4.10.

Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:291]: Confirmed with Jan that AliCloud is GA for 4.10.

**Comment by duanwei33** [comment]
Hi @lpettyjo 
looks like this PR is not included which need fix the conflicts.
https://github.com/openshift/openshift-docs/pull/41656/files

File: release_notes/ocp-4-10-release-notes.adoc [hunk 16]
```diff
@@ -286,6 +286,75 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
...
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Azure Disk. This feature was previously introduced as a Technology Preview feature in {product-title} 4.8 and is now generally available and enabled by default in {product-title} 4.10.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure.adoc#persistent-storage-csi-azure-disk[Azure Disk CSI Driver Operator].
+
+[id="ocp-4-10-storage-aws-efs-csi-operator"]
+==== Persistent storage using AWS Elastic File Storage CSI Driver Operator is generally available
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for AWS Elastic File Storage (EFS). This feature was previously introduced as a Technology Preview feature in {product-title} 4.9 and is now generally available and enabled by default in {product-title} 4.10.
+
```
Comment (chao007) [inline release_notes/ocp-4-10-release-notes.adoc:325]: EFS csi operator is managed by OLM.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 17]
```diff
@@ -286,6 +286,75 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
...
+==== Using fsGroup to reduce pod timeouts
+
+If a storage volume contains many files (~1,000,000 or greater), you may experience pod timeouts.
+
+{product-title} 4.10 introduces the ability to use `fsGroup` and `fsGroupChangePolicy` to skip recursive permission change for the storage volume, thus helping to avoid pod timeout problems.
+
+For more information, see xref:../storage/persistent_storage/understanding-persistent-storage.adoc#storage-persistent-storage-fsGroup[Using fsGroup to reduce pod timeouts].
+
+[id="ocp-4-10-storage-shared-dirs-local-vols"]
+==== Using shared directories with local volumes
```
Comment (duanwei33) [inline release_notes/ocp-4-10-release-notes.adoc:352]: @lpettyjo @gnufied 
Does this worth a new feature here?

File: release_notes/ocp-4-10-release-notes.adoc (hunk 15 above)
Comment (jsafrane) [reply release_notes/ocp-4-10-release-notes.adoc:291]: Alibaba is GA.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 18]
```diff
@@ -286,6 +286,75 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
...
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. This feature now supports automatic migration for the Azure File in-tree plug-in to the Azure File CSI driver.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc#persistent-storage-csi-migration[CSI automatic migration].
+
+[id="ocp-4-10-storage-csi-migration-vsphere"]
+==== Automatic CSI migration supports VMware vSphere (Technology Preview)
+
+Starting with {product-title} 4.8, automatic migration for in-tree volume plug-ins to their equivalent Container Storage Interface (CSI) drivers became available as a Technology Preview feature. This feature now supports automatic migration for the vSphere in-tree plug-in to the vSphere CSI driver.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc#persistent-storage-csi-migration[CSI automatic migration].
```
Comment (jsafrane) [inline release_notes/ocp-4-10-release-notes.adoc:340]: We should perhaps emphasize somehow that:

* This requires vSphere 6.7u3 (+ link to "CSI driver installation on vSphere clusters" above).
* This requires users to remove non-Red Hat vSphere CSI driver (+ link to https://github.com/openshift/openshift-docs/pull/40571)
* Cluster will still be upgraded if the conditions above are not met, however, users should fix the above to get a supported vSphere CSI driver.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 17 above)
Comment (gnufied) [reply release_notes/ocp-4-10-release-notes.adoc:352]: It always worked. We should probably remove it from release notes.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 16 above)
Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:566]: I've removed the statement about being enabled by default.

**Comment by lpettyjo** [comment]
> Hi @lpettyjo looks like this following PR is not included which need fix the conflicts. https://github.com/openshift/openshift-docs/pull/41656/files

It looks like that PR is merged now.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 17 above)
Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:593]: @duanwei33 @gnufied There is part of "fsGroup" and part of "shared local volumes" in this reference. Which one did you want removed?

File: release_notes/ocp-4-10-release-notes.adoc (hunk 18 above)
Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:581]: See my changes for this topic. Let me know if Ok.

Comment (jsafrane) [reply release_notes/ocp-4-10-release-notes.adoc:581]: I am sorry, I put my comment on a wrong place. Can you please move it to "Persistent storage using VMware vSphere CSI Driver Operator is generally available"? It's not related to CSI migration that much.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 17 above)
Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:601]: I've deleted the "shared local vols" entry as per our conversation.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 18 above)
Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:581]: Fixed!

**Comment by jsafrane** [comment]
lgtm from my side

File: release_notes/ocp-4-10-release-notes.adoc [hunk 19]
```diff
@@ -536,6 +536,76 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
...
+[id="ocp-4-10-storage-vsphere-csi-operator"]
+==== Persistent storage using VMware vSphere CSI Driver Operator is generally available
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for vSphere. This feature was previously introduced as a Technology Preview feature in {product-title} 4.8 and is now generally available and enabled by default in {product-title} 4.10.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere[vSphere CSI Driver Operator].
+
+vSphere CSI Driver Operator installation requires:
+
+* vSphere 6.7u3 or later (xref:../installing/installing_vsphere/preparing-to-install-on-vsphere.adoc#installation-vsphere-infrastructure_preparing-to-install-on-vsphere[VMware vSphere CSI Driver Operator requirements])
```
Comment (duanwei33) [inline release_notes/ocp-4-10-release-notes.adoc:566]: @lpettyjo  Should we list all the requirement as same as in https://deploy-preview-38258--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-installation-vsphere-csi?

- Virtual hardware version 15 or greater
- vSphere version 6.7 Update 3 or greater
- VMware ESXi version 6.7 Update 3 or greater

Comment (duanwei33) [reply release_notes/ocp-4-10-release-notes.adoc:566]: others are all good to me.

Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:566]: Linked to the spot you indicate.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 20]
```diff
@@ -557,6 +557,76 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
...
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for vSphere. This feature was previously introduced as a Technology Preview feature in {product-title} 4.8 and is now generally available and enabled by default in {product-title} 4.10.
+
+For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere[vSphere CSI Driver Operator].
+
+vSphere CSI Driver Operator installation requires:
+
+* Certain minimum component versions installed. See xref:../release_notes/ocp-4-10-release-notes.adoc#ocp-4-10-installation-vsphere-csi[CSI driver installation on vSphere clusters]
+
+* Removal of any non-Red Hat vSphere CSI driver (xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-install-issues_persistent-storage-csi-vsphere[Removing a non-Red Hat vSphere CSI Operator Driver])
```
Comment (duanwei33) [inline release_notes/ocp-4-10-release-notes.adoc:589]: @lpettyjo This link did not mention anything about how to remove non-Red Hat vSphere CSI driver or related info, could we remove this link? 
xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-install-issues_persistent-storage-csi-vsphere

Comment (lpettyjo) [reply release_notes/ocp-4-10-release-notes.adoc:589]: @duanwei33 I'm not sure I understand this comment. That link does provide info about the implications of having a non-RH driver and a brief procedure on how to remove them:  

"To uninstall the non-Red Hat vSphere CSI Driver:

Delete the non-Red Hat vSphere CSI Driver (VMware vSphere Container Storage Plug-in) Deployment and Daemonset objects.

Delete the configmap and secret objects that were installed previously with the non-Red Hat vSphere CSI Driver.

Delete the non-Red Hat vSphere CSI driver CSIDriver object:"

Comment (duanwei33) [reply release_notes/ocp-4-10-release-notes.adoc:589]: @lpettyjo Oh my fault, looks like I got the wrong link last time, yes, this link does provide the correct info, really sorry for that.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 21]
```diff
@@ -557,6 +557,76 @@ For more information, see xref:../networking/metallb/about-metallb.adoc#about-me
...
+
```
[the remark this answers is not in the document]
Comment (RomanBednar) [reply release_notes/ocp-4-10-release-notes.adoc:640]: ack - looks good, thanks!

**Pull request merged**